rollup/fees/rollup_fee.go

```diff
@@ -173,9 +162,58 @@ func readGPOStorageSlots(addr common.Address, state StateDB) gpoState {
 	gpoState.l1BlobBaseFee = state.GetState(addr, rcfg.L1BlobBaseFeeSlot).Big()
 	gpoState.commitScalar = state.GetState(addr, rcfg.CommitScalarSlot).Big()
 	gpoState.blobScalar = state.GetState(addr, rcfg.BlobScalarSlot).Big()
+	gpoState.penaltyThreshold = state.GetState(addr, rcfg.PenaltyThresholdSlot).Big()
+	gpoState.penaltyFactor = state.GetState(addr, rcfg.PenaltyFactorSlot).Big()
 	return gpoState
 }
 
+// estimateTxCompressionRatio estimates the compression ratio for transaction data using da-codec
+// compression_ratio(tx) = size(tx) * PRECISION / size(zstd(tx))
+func estimateTxCompressionRatio(data []byte, blockNumber uint64, blockTime uint64, config *params.ChainConfig) *big.Int {
+	if len(data) == 0 {
+		return rcfg.Precision
+	}
+
+	// Compress data using da-codec
+	compressed, err := encoding.CompressScrollBatchBytes(data, blockNumber, blockTime, config)
```

Comment:
So the goal of this change (passing config etc) is that we can pick different zstd version and compression params for different protocol versions?